src/nifs.rs

```diff
@@ -67,7 +70,52 @@ impl<E: Engine> NIFS<E> {
     let U = U1.fold(U2, &comm_T, &r);
 
     // fold the witness using `r` and `T`
-    let W = W1.fold(W2, &T, &r)?;
+    let W = W1.fold(W2, &T, &r_T, &r)?;
+
+    // return the folded instance and witness
+    Ok((Self { comm_T }, (U, W)))
+  }
+
+  /// Same as `prove`, but takes two Relaxed R1CS Instance/Witness pairs
+  pub fn prove_relaxed(
```

Comment:
Nit: It might be easier to introduce `NIFSRelaxed` type and have `prove` and `verify` methods on that type. It will eliminate someone from calling `verify_relaxed` when the prover has called `prove` and vice versa. What do you think?

Reply:
Sounds like a good idea! will do.

Reply:
I think it still makes sense to derandomize outside NIFSRelaxed, to avoid confusion.

(Also, absorbing both U1 and U2 in the RO is a standard version, not specialized to the case of a randomized instance. Isn't it?)

Reply:
will do.

I guess I just meant it doesn't incorporate the previous optimization that ignores the U1 absorption, since U2.X[0] = Hash(params, U1, i, z0, zi).